test/integration/puppetlabs/services/jruby/jruby_locking_test.clj

```diff
@@ -2,93 +2,99 @@
   (:require [clojure.test :refer :all]
             [puppetlabs.services.jruby.jruby-testutils :as jruby-testutils]
             [puppetlabs.trapperkeeper.app :as tk-app]
-            [puppetlabs.services.jruby.jruby-service :as jruby-service]
-            [puppetlabs.services.protocols.jruby :as jruby-protocol]
             [schema.test :as schema-test]
+            [puppetlabs.services.jruby.jruby-core :as jruby-core]
             [puppetlabs.trapperkeeper.testutils.bootstrap :as tk-bootstrap]
-            [puppetlabs.trapperkeeper.core :as tk]))
+            [puppetlabs.services.protocols.pool-manager :as pool-manager-protocol]))
 
 (use-fixtures :once schema-test/validate-schemas)
 
-(defn jruby-service-test-config
+(defn jruby-test-config
   [pool-size]
-  (jruby-testutils/jruby-tk-config
-    (jruby-testutils/jruby-config {:max-active-instances pool-size
-                                          :borrow-timeout 1})))
+  (jruby-testutils/jruby-config {:max-active-instances pool-size
+                                 :borrow-timeout 1}))
 
 (defn can-borrow-from-different-thread?
-  [jruby-service]
+  [pool-context]
   @(future
-    (if-let [instance (jruby-protocol/borrow-instance jruby-service :test)]
+    (if-let [instance (jruby-core/borrow-from-pool-with-timeout pool-context :test [])]
       (do
-        (jruby-protocol/return-instance jruby-service instance :test)
+        (jruby-core/return-to-pool instance :test [])
         true))))
 
 (deftest ^:integration with-lock-test
   (tk-bootstrap/with-app-with-config
-    app
-    jruby-testutils/default-services
-    (jruby-service-test-config 1)
-    (jruby-testutils/wait-for-jrubies app)
-    (let [jruby-service (tk-app/get-service app :JRubyService)]
-      (testing "initial state of write lock is unlocked"
-        (is (can-borrow-from-different-thread? jruby-service))
-      (testing "with-lock macro holds write lock while executing body"
-        (jruby-service/with-lock jruby-service :with-lock-holds-lock-test
-          (is (not (can-borrow-from-different-thread? jruby-service)))))
-      (testing "with-lock macro releases write lock after exectuing body"
-        (is (can-borrow-from-different-thread? jruby-service)))))))
+   app
```

Comment:
There's a lot of whitespace changes in this PR that got mixed in with everything. I tried to minimize that, but it got really hard. Sorry. :( Maybe I have my intellij settings wrong, or our standards are different from when this code was first written. I realized after the fact that probably I should have tried to do the whitespace changes first, but by the time I realized that it was kind of too late.